vocab.py

```diff
@@ -20,9 +28,92 @@ class ValidDataSourceName(Enum):
     Each data source name should be defined in ValidDataSourceName.
     NAME_OF_DATA_SOURCE = auto()
     """
-    
+
     CANVAS_API = auto()
     CANVAS_ZOOM_MEETINGS = auto()
     KALTURA_API = auto()
     UNIZIN_DATA_PLATFORM_EVENTS = auto()
     UNIZIN_DATA_WAREHOUSE = auto()
+
+
+class DataSourceStatus:
+    '''
+    Each data source process must return an instance of this class, which tells the name of the
+    data source and the time at which data from it was last updated.
+    '''
+    def __init__(
+            self,
+            dataSourceName: ValidDataSourceName = NotImplemented,
+            dataUpdatedAt: datetime = NotImplemented
+    ) -> None:
+        '''
+
+        :param dataSourceName: `ValidDataSourceName` - Name of data source
+        :param dataUpdatedAt: `datetime` - Time at which data from the source was updated
+        '''
+        self.dataSourceName: ValidDataSourceName
+        self.dataUpdatedAt: datetime
+        self.setDataSourceName(dataSourceName)
+        self.setDataUpdatedAt(dataUpdatedAt)
+
+    def setDataSourceName(
```

Comment:
The preferred way to implement these in python is to use a [property decorator](https://docs.python.org/3/library/functions.html#property) and the getter/setter decorators for those.

Reply:
Thanks for the suggestion.  I've not done it that way before, but I'll give it a try.

I've looked at the example you linked to.  I'm unsure whether in the end the methods are intended to be called as `instOfC.x = 5` and `instOfC.x` for set and get, respectively.  I understand that some may want to use properties that way, but I don't like it.  It also doesn't make a fluent interface possible.

Reply:
Yeah, that's basically how you'd do it.

[Python is not Java
](https://docs.quantifiedcode.com/python-anti-patterns/correctness/implementing_java-style_getters_and_setters.html)

Even in Java everything uses Lombok annotations now for these.

Reply:
I never thought I'd say it:  I'm starting to dislike Python.

Reply:
I'm going to take a hybrid approach.  I will follow the Pythonic techniques to define functions for getting and setting the properties.  However, I will keep the setter properties so the class can retain a fluent interface.

I believe Guido has said he's not in favor of fluent interfaces.  But I like them.  Even though he is BDFL, this is still a democracy.